xbmc/addons/Repository.cpp

```diff
@@ -201,15 +201,25 @@ bool CRepository::FetchIndex(const RepositoryDirInfo& repo,
                              std::string const& digest,
                              std::vector<AddonInfoPtr>& addons) noexcept
 {
-  XFILE::CCurlFile http;
-
   std::string response;
-  if (!http.Get(repo.info, response))
+  CFile file;
+  if (!file.Open(repo.info))
   {
     CLog::Log(LOGERROR, "CRepository: failed to read {}", repo.info);
     return false;
   }
 
+  // we intentionally avoid using file.GetLength() for
+  // Transfer-Encoding: chunked servers.
+  std::stringstream ss;
+  char temp[1024];
+  int read;
+  while ((read = file.Read(temp, sizeof(temp))) > 0)
+    ss.write(temp, read);
+  if (read <= -1)
+    return false;
+  response = ss.str();
```

Comment:
To avoid copying the string into `response`:
```suggestion
  response = std::move(ss).str();
```

Reply:
@neo1973 I solved the problem via different set of files so your `std::move` suggestions have fallen out of scope for this PR.

But I have prepared a new PR just for that: #25728